### ISSUE TITLE
Event fired hours after initiating. 

### ISSUE THREAD

**Issue by benvallack** [Kemmey/Kemmey-TeslaWatch-Public#262]
My watch had no signal when I tried to open my charge port. I forgot about it then several hours later I heard the watch make the success sound and my charge port opened! Clearly I'd rather this didn't behave like this! Surely it should give up if it can't connect in 10 seconds or so?

**Comment by Kemmey** [comment]
@benvallack Yeah, absolutely should. After 40 seconds, in fact. 

I'll see if I can reproduce, that definitely is not intended behavior.

**Comment by benvallack** [comment]
Ah ok sure. Let me know if there are any logs I can send from the app to help.

**Comment by Kemmey** [comment]
Hey @benvallack 

Sorry for the long follow-up delay - I've been fighting this all through development of 1.1.6 - and I've completely rewritten the backgrounding code multiple times. There are two scenarios under which this was happening - I believe I've solved the worst offender, which was during wake - so that should now happen properly in the background and fail after about 40 seconds and definitely not try to run again. The other scenario is if your watch/phone/iPad loses connectivity when sending the command - then the OS will pick up that request whenever connectivity is resumed - that's a bit more tricky, so I'm still working on that. 

Do let me know if you continue to experience this rather dangerous bug after the release of 1.1.6. 

Cheers, 

Kim

**Comment by hkunzig** [comment]
I have the same issue. V1.1.6 build 534.  It’s connectivity for me, not waking. I had my climate come on the other night 3 hours after I asked it to start. The only way I knew was because the tesla app notified me at 5am that it was shutting it off.  Just now I had the trunk close on my head about 5 minutes after it failed to open. Thankfully it did since it’s pouring out and I didn’t think I could go back in the building until it actually issued. I like the app, but this issue really makes it unusable for me. I primarily want it so I can tell siri to open the trunk when my hands are full.

**Comment by Kemmey** [comment]
@hkunzig wow - not good! :-( 

Thanks for letting me know! 

You mention that in your case it is connectivity - do you mean to say that you're certain the app has woken up the car when you issue the command, but that connectivity to your watch is spotty - so the network request sending the actual command gets prolonged? 

I'll continue to test this further - and have myself experienced this once since release. So the issue certainly is still present. 

Do you encounter this frequently? Are you able to somewhat reliably reproduce?

**Comment by hkunzig** [comment]
Yes, I'm certain the car is awake.  The most recent time I was on fringe wifi for my phone and the watch was out of wifi range.  My watch is wifi only.  I used the complication to open the trunk.  It didn't do anything before I got to the car, so I opened it manually.  I pulled out my phone and saw it was fringe wifi, so I turned off the wifi.  Shortly after that the trunk closed.  Wifi assist is on, but I don't think it works too well.   I don't encounter it very frequently; usually the problem is the watch or siri so it fails immediately :)    My favorite is when siri randomly can't find a shortcut even though it displays the name of it perfectly.  This one I could reproduce easily.

The issue with the climate is a little different.  The car was definitely awake; sentry was on.  I was leaving a restaurant and was not on wifi.  I did it through siri vs the complication.  I didn't actually realize it didn't work until the next morning when I saw the alert that the climate had been on for 4 hours.  That was around 5am.  I had issued the request around 10pm.

Using the complication for the trunk; the watch app sounds a tone when the trunk opens, not when the command is issued.  Not sure if that helps at all

Thanks

**Comment by Kemmey** [comment]
@hkunzig thanks - that was very detailed and helpful! This sounds to me exactly like the second scenario, where as you stated, the car is properly awake, but the command execution fails to complete while connected, so the OS will keep the request around and fire it off when back online. This is extremely hard for me to control as from my end, I've already marked that request as cancelled - but the OS keeps it around and fires it anyway. But at least I know where my bug is - I must not mark it as cancelled properly, since the OS eventually fires it off anyway. 

I'll keep digging - thank you for reporting on this!

**Comment by GAW44** [comment]
I made a request on my watch and nothing happened.  Two days later I received a notification that the old request has now happened.  In turn, I had to go outside and close my frunk.

**Comment by Kemmey** [comment]
@GAW44 which version of the app did this happen with? 1.1.7 should be protected against this, to the best of my knowledge. If not I'd be extremely interested in the logs.

**Comment by GAW44** [comment]
1.1.7



Greg

**Comment by Kemmey** [comment]
Hey @GAW44 

Ok, so this is happening when invoking Siri? Not after tapping a button or running a shortcut?

You asked Siri to pop the trunk, nothing happened, and two days later, Siri responds that it popped the trunk?

Or did you get a notification from watch app that the trunk had successfully been opened?

Just trying to narrow this one down.

**Comment by GAW44** [comment]
Hard to tell.  I tried with Siri and nothing happened and then I used the shortcut, both didn't work.  

I got a notification on my phone a few days later that my trunk was successfully opened.
